src/main/resources/com/google/api/codegen/csharp/gapic_client.snip

```diff
@@ -0,0 +1,465 @@
+    // you may not use this file except in compliance with the License.
+    // You may obtain a copy of the License at
+    //
+    //     http://www.apache.org/licenses/LICENSE-2.0
+    //
+    // Unless required by applicable law or agreed to in writing, software
+    // distributed under the License is distributed on an "AS IS" BASIS,
+    // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    // See the License for the specific language governing permissions and
+    // limitations under the License.
+
+    // Generated code. DO NOT EDIT!
+
+    @join import : common.api.imports
+        using {@import.fullName};
+    @end
+
+    namespace {@common.api.packageName}
+    {
+        {@settings(common.api, common.settings)}
+
+        {@abstractClient(common.api, common.settings)}
+
+        {@implClient(common.api, common.settings)}
+        
+        {@partialPageStreaming(common.api, common.settings)}
+    }
+{@""}
+@end
+
+@private settings(api, settings)
+    /// <summary>
+    /// Settings for a <see cref="{@api.name}"/>.
+    /// </summary>
+    public sealed partial class {@settings.name} : ServiceSettingsBase
+    {
+        /// <summary>
+        /// Get a new instance of the default <see cref="{@settings.name}"/>.
+        /// </summary>
+        /// <returns>
+        /// A new instance of the default <see cref="{@settings.name}"/>.
+        /// </returns>
+        public static {@settings.name} GetDefault() => new {@settings.name}();
+
+        /// <summary>
+        /// Constructs a new <see cref="{@settings.name}"/> object with default settings.
+        /// </summary>
+        public {@settings.name}() { }
+        
+        private {@settings.name}({@settings.name} existing) : base(existing)
+        {
+            GaxPreconditions.CheckNotNull(existing, nameof(existing));
+            @join call : settings.callSettings
+                {@call.methodName}Settings = existing.{@call.methodName}Settings?.Clone();
+            @end
+        }
+        
+        @join retry : settings.retryCodesDefinitions
+            /// <summary>
+            /// The filter specifying which RPC <see cref="StatusCode"/>s are eligible for retry
+            /// for "{@retry.name}" <see cref="{@api.name}"/> RPC methods.
+            /// </summary>
+            /// <remarks>
+            @if retry.codes
+               /// The eligible RPC <see cref="StatusCode"/>s for retry for "{@retry.name}" RPC methods are:
+               /// <list type="bullet">
+               @join code : retry.codeNames
+                   /// <item><description><see cref="StatusCode.{@code}"/></description></item>
+               @end
+               /// </list>
+            @else
+                /// There are no RPC <see cref="StatusCode"/>s eligible for retry for "{@retry.name}" RPC methods.
+            @end
+            /// </remarks>
+            public static Predicate<RpcException> {@retry.name}RetryFilter { get; } =
```

Comment:
This new approach definitely goes in a direction away from the single-language point of view. It turns out that most people on the vkit team aren't working on just one language - they often deal with 2 or 3 regularly, and occasionally all of the rest (with certain features that we do for all languages). That was actually one of the (multiple) motivations for orienting the design more in this direction. Our four dynamic languages have quite a bit of pieces in common, and three of them were actually bootstrapped off of the same starting point (Python). 

Now maybe it's true that banning *all* identifier composition in the templates might end up being extreme. I think it's worked out ok so far; Java and PHP were written this way, and Go is about to be merged following this pattern. In any case, I kind of want to follow this particular experiment to its conclusion, despite being a little bit non-optimal in certain local cases. 

My hypothesis is that this approach would make it easier (and safer) for non-C# developers to make certain changes or additions, because they could depend on the system to make it safe for them. One particular example is enforcement of input data casing; we had some weird naming problems earlier because we assumed that certain input data was lower-underscore when it was actually lower-camel, producing really weird names for Python. When we go through the Name class, all input name casing is enforced, so we catch it at generation time. In your case, what happens if you want to form a variable name out of retry.name? You have to be aware that it is currently in upper camel, and then you would have to convert it to lower camel in the template, and then you're back into template casing hell again. With the new casing system, our snippet templates are cut down in size by about 30-50% (depending on the template). The short of it is that I'm currently wary of touching the current C# generation, but I would feel much less concern about touching it under the MVVM refactoring with no composition in the templates. I don't want the C# generation to be siloed away from the rest; it's under the ownership of the larger toolkit umbrella. 

Another reason I'm pushing for zero composition in the templates is that I'm concerned with a broken windows situation. It's usually a smaller initial effort compose in the templates, and if there are already cases where it's done in a template, it's not a big leap to add even more. 

Now all that said, if the experiment doesn't work out, I'm happy to back off on the requirement.